app.go

```diff
@@ -138,7 +138,13 @@ type App struct {
 
 // New creates and initializes an App. All applications begin with the
 // Lifecycle type available in their dependency injection container.
+//
+// It then executes all functions supplied via the Invoke option. Supplying
+// arguments to these functions requires calling some of the constructors
+// supplied by the Provide option. If any invoked function fails, an error is
+// returned immediately.
 func New(opts ...Option) *App {
+	// TODO(glib): New should probably return (*App, error)
```

Comment:
Why? 
1. It is so much easier to follow the same pattern as the constructors.
2. Panic means that something went really wrong, so we should do it when the internal logic is inconsistent, users invoke's can easily return errors.

Another solution could be provide to an option, to skip/panic/exit on errors(like pflag: https://github.com/spf13/pflag/blob/master/flag.go#L119)